### PR TITLE
fix: Validate image size for Stable Diffusion 3 on adapter side

### PR DESCRIPTION
**Pull request by roman-romanov-o**

### CODE REVIEW

File: aidial_adapter_bedrock/llm/model/stability/v2.py [hunk 1]
```diff
@@ -27,10 +30,57 @@
...
+    image: Resource,
+    width_constraints: Tuple[int, int] | None,
+    height_constraints: Tuple[int, int] | None,
+) -> None:
+    if width_constraints is None and height_constraints is None:
+        return
+
+    with Image.open(BytesIO(image.data)) as img:
+        width, height = img.size
+        if width_constraints and not (
```
Comment (adubovik) [inline aidial_adapter_bedrock/llm/model/stability/v2.py]: Would have been great to reduce code duplication:

```python
for rng, val, name in [(width_constraints, width, "width"), (height_constraints, height, "height")]:
    if rng is None:
        continue
    (min_val, max_val) = val
    ...
```

File: aidial_adapter_bedrock/llm/model/stability/v2.py [hunk 2]
```diff
@@ -27,10 +30,57 @@
...
 
 SUPPORTED_IMAGE_TYPES = ["image/jpeg", "image/png", "image/webp"]
 
 
+def _validate_image_size(
+    image: Resource,
+    width_constraints: Tuple[int, int] | None,
+    height_constraints: Tuple[int, int] | None,
+) -> None:
+    if width_constraints is None and height_constraints is None:
```
Comment (adubovik) [inline aidial_adapter_bedrock/llm/model/stability/v2.py:43]: redundant shortcut, it doesn't save much machine cycles, wastes human cycles.

File: aidial_adapter_bedrock/llm/model/stability/v2.py (hunk 1 above)
Comment (roman-romanov-o) [reply aidial_adapter_bedrock/llm/model/stability/v2.py]: Good idea,done

**Comment by roman-romanov-o** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-adapter-bedrock-pr-175.nightly-test.deltixhub.io
>E2E tests status: failed

**Review by adubovik — APPROVED** [review]

**Pull request merged**